### PR TITLE
Add docs around package scanning and notify user of infected package upload.

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

https://jira.autodesk.com/browse/DYN-4691
https://jira.autodesk.com/browse/DYN-4695

This PR adds documentation around package scanning, and now users with infected packages upload will see dismissible toast notifications when loading package manager search window.

**Documentation:**
devenv_eYuSnKSoX8

**Infected package toast notification with package download:-**
Revit_w9lfRphlwW

**SS of notification:**
Screen Shot 2022-08-22 at 5 37 24 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Add documentation around package scanning
Add notifications for users with infected package uploads

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 1]
```diff
@@ -471,6 +472,28 @@ public List<PackageManagerSearchElement> ListAll()
...
+        /// Returns a dictionary of infected package(s) with name and version, if the last published version of package uploaded by the current user was flagged as infected.
+        /// </summary>
+        /// <returns></returns>
+        public List<PackageManagerSearchElement> GetInfectedPackages()
+        {
+            InfectedPackageList = new List<PackageManagerSearchElement>();
+            var latestPkgs = Model.GetUsersLatestPackages();
+            if (latestPkgs != null && latestPkgs.maintains.Count > 0)
+            {
+                foreach (var infectedVer in latestPkgs.maintains)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:485]: Would you remind me why do we need to look at each version here? I thought we only need to check the latest version of each package

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 2]
```diff
@@ -471,6 +472,28 @@ public List<PackageManagerSearchElement> ListAll()
...
+                {
+                    if (infectedVer.scan_status == "quarantined")
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:487]: Would you define this state string as static somewhere?

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs [hunk 3]
```diff
@@ -543,17 +549,28 @@ public void ClearSearchTextBox(object obj)
         /// one of these toast notifications.
         /// </summary>
         /// <param name="obj"></param>
-        public void ClearDownloadToastNotification(object obj)
+        public void ClearToastNotification(object obj)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs:552]: This is API breaking but I guess you decided to break it intentionally?

Comment (zeusongit) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs:552]: Yes to make this function a bit more generic

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs (hunk 1 above)
Comment (zeusongit) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:485]: The API will return all packages with last uploaded version, so 1 version per package only.

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml [hunk 4]
```diff
@@ -930,20 +1090,20 @@
                                     FontFamily="{StaticResource ArtifaktElementRegular}"
                                     FontSize="10"
                                     Foreground="#F5F5F5"
-                                    Text="{Binding Path=DownloadState, Converter={StaticResource PackageDownloadStateToStringConverter}}" />
+                                    Text="Upload Error" />
 
                                 <!--  Version  -->
                                 <TextBlock
-                                    Name="version"
+                                    Name="iversion"
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml:1097]: What does `iversion` mean?

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml [hunk 5]
```diff
@@ -930,20 +1090,20 @@
                                     FontFamily="{StaticResource ArtifaktElementRegular}"
                                     FontSize="10"
                                     Foreground="#F5F5F5"
-                                    Text="{Binding Path=DownloadState, Converter={StaticResource PackageDownloadStateToStringConverter}}" />
+                                    Text="Upload Error" />
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml:1093]: Resource string?

**Review by QilongTang — COMMENTED** [review]
LGTM with some comments

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml (hunk 4 above)
Comment (zeusongit) [reply src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml:1097]: `infected version` just a name for the text block, no relation to apple 😛

**Review by QilongTang — APPROVED** [review]

**Pull request merged**